packages/sharing-smart-contract/test/e2e/consumeProtectedData.test.js

```diff
@@ -137,47 +137,138 @@ describe('ConsumeProtectedData', () => {
     });
 
-    describe('voucher - consumeProtectedData()', () => {
-      it('should create a deal on chain and consume asset in the user voucher', async () => {
-        const {
-          dataProtectorSharingContract,
-          pocoContract,
-          protectedDataAddress,
-          appAddress,
-          collectionTokenId,
-          subscriptionParams,
-        } = await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
-        const { voucherOwner, workerpoolOrder } = await createVoucher(await dataProtectorSharingContract.getAddress());
-
-        await pocoContract
-          .connect(voucherOwner)
-          .approve(await dataProtectorSharingContract.getAddress(), subscriptionParams.price);
-        await pocoContract.connect(voucherOwner).deposit({
-          value: ethers.parseUnits(subscriptionParams.price.toString(), 'gwei'),
-        }); // value sent should be in wei
+    // TODO: when _spender has not approved the contract => should revert the consumeProtectedData tx
+    it('should revert if the consumer has not approved the DataProtectorSharingContract', async () => {
+      const { dataProtectorSharingContract, protectedDataAddress, appAddress, collectionTokenId, subscriptionParams } =
+        await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
+      const dataProtectorSharingAddress = await dataProtectorSharingContract.getAddress();
+      const { voucherOwner, workerpoolOrder } = await createVoucher({ dataProtectorSharingAddress });
+
+      await dataProtectorSharingContract
+        .connect(voucherOwner)
+        .subscribeToCollection(collectionTokenId, subscriptionParams);
 
-        await dataProtectorSharingContract
+      expect(
+        dataProtectorSharingContract
           .connect(voucherOwner)
-          .subscribeToCollection(collectionTokenId, subscriptionParams);
+          .consumeProtectedData(protectedDataAddress, workerpoolOrder, appAddress, true),
+      ).to.be.reverted;
+    });
+  });
+
+  describe('voucher - consumeProtectedData()', () => {
+    it('should create a deal on chain and consume asset in the user voucher', async () => {
+      const {
+        dataProtectorSharingContract,
+        pocoContract,
+        protectedDataAddress,
+        appAddress,
+        collectionTokenId,
+        subscriptionParams,
+      } = await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
+      const dataProtectorSharingAddress = await dataProtectorSharingContract.getAddress();
+      const { voucherOwner, workerpoolOrder } = await createVoucher({ dataProtectorSharingAddress });
+
+      await pocoContract
+        .connect(voucherOwner)
+        .approve(await dataProtectorSharingContract.getAddress(), subscriptionParams.price);
+      await pocoContract.connect(voucherOwner).deposit({
+        value: ethers.parseUnits(subscriptionParams.price.toString(), 'gwei'),
+      }); // value sent should be in wei
```

Comment:
the deal is supposed to be paid by the voucher, the user doesn't need anything on its account neither to approve the sharing contract to use its account